File: Stack.py
class node:
          def __init__(self,nextnode,val):
                    
                    self.nextnode=nextnode
                    self.val=val

class stack:
          def __init__(self,top,size):
                     
                    self.top=top
                    self.size=size
          
          def push(self,value):
                    
                    n=node(None,value)
                    n.nextnode=self.top
                    self.top=n
                    self.size+=1

          def pop(self):
                    
                    if(self.top==None):return None
                    n=self.top
                    self.top=n.nextnode
                    self.size-=1
                    return n

          def peek(self):
                    if(self.top==None):return None
                    return self.top.val

letters=['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
string=str(input("Give a string to find it's reverse\n"))
print(string,'\n')

S=stack(None,0)

for i in range(len(string)):
          S.push(string[i])
          print(S.peek())
          
print("\n\n")

for i in range(len(string)):
          print(S.pop().val)
 

File: Stack.cpp
#include<stdio.h>
#include<stdlib.h>
struct node
{
	int val;
	struct node *next;
};

struct stack
{
	int size;
	struct node *top;
};

bool isempty(struct stack *s)
{
	return(s->size==0);
};

void push(struct stack *s,int i)
{
	struct node *n=(struct node*)malloc(sizeof(struct node));
	n->val=i;
	n->next=s->top;
	s->top=n;
	(s->size)++;
}

struct node pop(struct stack *s)
{
	struct node *n=(struct node*)malloc(sizeof(struct node));
	if(isempty(s))
	{
		return(*n);
	}
	n=s->top;
	s->top=s->top->next;
	return(*n);
}

int peek(struct stack *s)
{	
	return(s->top->val);
}

int main()
{	
	struct stack *s=(struct stack*)malloc(sizeof(struct stack));
	s->size=0;
	push(s,2);
	printf("%d",peek(s));
}
  